scheds/rust/scx_chaos/src/bpf/main.bpf.c

```diff
@@ -452,6 +463,21 @@ void BPF_STRUCT_OPS(chaos_enqueue, struct task_struct *p __arg_trusted, u64 enq_
 	    enqueue_chaotic(p, enq_flags, taskc))
 		return;
 
+	// NOTE: this may not work for affinitized tasks because p2dq does
+	// direct dispatch in some situations.
+	if (taskc->next_trait == CHAOS_TRAIT_DEGRADATION && degradation_percent > 0) {
+		if (promise.kind == P2DQ_ENQUEUE_PROMISE_FIFO) {
+			promise.fifo.slice_ns = ((100 - degradation_percent) * promise.fifo.slice_ns) / 100;
```

Comment:
could we switch these to `/ 128` so it's a bitshift? Would change the var name to `_frac7` but it's basically a percentage.